### ISSUE TITLE
check impact of upgrading lightning 2.1 -> 2.2

### ISSUE THREAD

**Issue by ArneBinder** [ArneBinder/pie-document-level#374]
especially, check "Improved Handling of Evaluation Mode" from the [lightning 2.2.0 release notes](https://github.com/Lightning-AI/pytorch-lightning/releases/tag/2.2.0)

context: the reproducibility expeiurments from [here](https://github.com/ArneBinder/pie-document-level/pull/366#issuecomment-2628973432) (and [here](https://github.com/ArneBinder/pie-document-level/pull/365#issuecomment-2634652739)) indicate, that some non-reproducibility is introduced when upgrading pytorch lightning from 2.1.2 to 2.4.0  (but note that updating torch from 2.0.0 to 2.6.0 has also an effect!)

**Comment by ArneBinder** [comment]
Check if adding this to the models has any effect:

```python
    def on_train_start(self) -> None:
        self.train()
```

**Comment by ArneBinder** [comment]
from another project with `lightning >=2.2.0` (and embedded Huggingface model):
```
Trainable params: 109 M                                                         
Non-trainable params: 0                                                         
Total params: 109 M                                                             
Total estimated model params size (MB): 439                                     
Modules in train mode: 28                                                       
Modules in eval mode: 230    
```
Note that `Non-trainable params: 0`, but `Modules in eval mode: 230`!
